contracts/masset/liquidator/Liquidator.sol

```diff
@@ -0,0 +1,263 @@
+    }
+
+
+    /**
+    * @dev Internal initialize function, to set the Uniswap address
+    * @param _uniswapAddress   Uniswap contract address
+    */
+    function _initialize(address _uniswapAddress)
+        internal
+    {
+        uniswapAddress = _uniswapAddress;
+    }
+
+    /**
+    * @dev Create a liquidation
+    * @param _bAsset The _bAsset address that this liquidation is for
+    * @param _integration The integration contract address for the _bAsset
+    * @param _sellToken The token address to be sold
+    * @param _uniswapPath The Uniswap path as an array of addresses e.g. [COMP, WETH, DAI]
+    * @param _amount The amount of tokens to be sold
+    * @param _paused Whether the liquidation is paused
+    */
+    function createLiquidation(
+        address     _bAsset,
+        address     _integration,
+        address     _sellToken,
+        address[]   calldata _uniswapPath,
+        uint        _amount,
+        bool        _paused
+    )
+        external
+        onlyGovernor
+    {
+        require(_bAsset != address(0), "_bAsset cannot be zero address");
+        require(_integration != address(0), "integration cannot be zero address");
+
+        Liquidation storage liq = liquidations[_bAsset];
+        liq.bAsset = _bAsset;
+        liq.integration = _integration;
+        liq.sellToken = _sellToken;
+        liq.uniswapPath = _uniswapPath;
+        liq.amount = _amount;
+        liq.paused = _paused;
+
+        emit LiquidationAdded(_bAsset);
+    }
+
+    /**
+     * @dev Get a liquidation
+     * @param   _bAsset Address for the underlying bAsset
+     * @return  liquidation The liquidation data
+     */
+    function readLiquidation(address _bAsset)
+        external
+        onlyGovernor
+        returns (Liquidation memory liquidation)
+    {
+        require(liquidations[_bAsset].bAsset != address(0), "No liquidation for this bAsset");
+
+        liquidation = liquidations[_bAsset];
+        return liquidation;
+    }
+
+    /**
+    * @dev Update a liquidation
+    * @param _bAsset The _bAsset address that this liquidation is for
+    * @param _integration The integration contract address for the _bAsset
+    * @param _sellToken The token address to be sold
+    * @param _uniswapPath The Uniswap path as an array of addresses e.g. [COMP, WETH, DAI]
+    * @param _amount The amount of tokens to be sold
+    * @param _paused Whether the liquidation is paused
+    */
+    function updateLiquidation(
+        address     _bAsset,
+        address     _integration,
+        address     _sellToken,
+        address[]   calldata _uniswapPath,
+        uint        _amount,
+        bool        _paused
+    )
+
+        external
+        onlyGovernor
+    {
+        require(liquidations[_bAsset].bAsset != address(0), "No liquidation for this bAsset");
+
+        Liquidation storage liq = liquidations[_bAsset];
+        liq = liquidations[_bAsset];
+        liq.bAsset = _bAsset;
+        liq.integration = _integration;
```

Comment:
We should probably only change this if there are no outgoing tokens still left to be collected

Reply:
Added a check for this, commented until I add a mock for tests.